lib/binary_search_tree.rb

```diff
@@ -0,0 +1,35 @@
+require_relative 'tree_node'
+
+class BinarySearchTree
+  attr_accessor :root
+
+  def insert(score, title)
+    new_node = TreeNode.new(score, title)
+
+    return set_root(new_node) if root.nil?
+
+    current_node = root
+    current_depth = 0
+
+    until current_node.movie_score == score
+      if score < current_node.movie_score
+        current_node.left = new_node if current_node.left.nil?
+        current_node = current_node.left
+      elsif score > current_node.movie_score
+        current_node.right = new_node if current_node.right.nil?
+        current_node = current_node.right
+      end
```

Comment:
I feel like this can be abstracted to a private method. A method that would be a little more versatile, just searching for the node and returning it. then you can make two simple guard statements asking if the new_node score is greater or less than the return of the search.